### PR TITLE
[issue#211] Remove use of Jakarta Deployment and classes that use it (no replacement deployment interfaces added)

### PR DESCRIPTION
**Pull request by scottmarlow**
https://github.com/eclipse-ee4j/jakartaee-tck/issues/211

### CODE REVIEW

**Comment by scottmarlow** [comment]
[Some interesting discussion about adding adapting the existing Arquillian servers (connectors)](https://www.eclipse.org/lists/jakartaee-platform-dev/msg01837.html) such that we could include [vendor specific deployment descriptors files or rely on vendors to add these files via an extension](https://www.eclipse.org/lists/jakartaee-platform-dev/msg01837.html).

File: src/com/sun/ts/lib/deliverable/cts/deploy/DeployTestUtil.java [hunk 1]
```diff
@@ -35,16 +35,9 @@
...
 import javax.enterprise.deploy.spi.status.*;
 import javax.enterprise.deploy.spi.exceptions.*;
-import javax.enterprise.deploy.model.*;
 import javax.enterprise.deploy.shared.*;
-import javax.enterprise.deploy.model.*;
-
-import java.io.*;
-import java.util.*;
 
 import javax.enterprise.deploy.shared.factories.*;
```
Comment (alwin-joseph) [inline src/com/sun/ts/lib/deliverable/cts/deploy/DeployTestUtil.java:40]: Are you planning to remove the pending occurrences of javax.enterprise.deploy also. Will that be done in another change ?

Comment (scottmarlow) [reply src/com/sun/ts/lib/deliverable/cts/deploy/DeployTestUtil.java:40]: I thought I removed all of the references, I will add an additional commit to remove the missed ones.  

I'll also remove the added wget of  https://repo1.maven.org/maven2/javax/enterprise/deploy/javax.enterprise.deploy-api/1.7/javax.enterprise.deploy-api-1.7.jar in docker/build_jakartaeetck.sh, as we shouldn't reference those classes any more.

Comment (scottmarlow) [reply src/com/sun/ts/lib/deliverable/cts/deploy/DeployTestUtil.java:40]: @alwin-joseph according to jakartaee/src/main/jbake/content/commonappdeploy.adoc, there are additional tests that depend on Jakarta Deployment, however, we removed Jakarta Deployment, so these tests will not work.
```
|Directory Under `com/sun/ts/tests` |Directory Under `com/sun/ts/tests`
With Associated Common Applications
|`ejb/ee/tx/session` |`ejb/ee/tx/txbean`

|`ejb/ee/tx/entity/bmp` |`ejb/ee/tx/txEbean`

|`ejb/ee/tx/entity/cmp` |`ejb/ee/tx/txECMPbean`

|`ejb/ee/tx/entity/pm` |`ejb/ee/tx/txEPMbean`

|`connector/ee/localTx/msginflow` |`common/connector/whitebox`

|`connector/ee/mdb` |`connector/ee/localTx`

|`common/connector/whitebox` |`connector/ee/noTx`

|`common/connector/whitebox` |`connector/ee/xa`

|`common/connector/whitebox` |`connector/ee/connManager`

|`common/connector/whitebox` |`xa/ee`

|`compat13/connector/localTx` |`compat13/connector/whitebox`

|`compat13/connector/noTx` |`compat13/connector/whitebox`

|`compat13/connector/xa` |`compat13/connector/whitebox`

|`interop/tx/session` |`interop/tx/txbean`

|`interop/tx/entity` |`interop/tx/txEbean`

|`interop/tx/webclient` |`interop/tx/txbean`

|`tests/interop/csiv2` |`interop/csiv2/rionly`

|`ejb/ee/pm/ejbql` |`ejb/ee/pm/ejbql/schema`

|`ejb/ee/tx/session/stateful/bm/TxMDBMS_Direct`
|`ejb/ee/tx/session/stateful/bm/TxMDBMSBeans/BeanA`

|`ejb/ee/tx/session/stateful/bm/TxMDBMSBeans/BeanB`
|`ejb/ee/tx/session/stateful/bm/TxMDBMSBeans/BeanC`

|`tests/ejb/ee/tx/session/stateful/bm/TxMDBMS_Indirect`
|`ejb/ee/tx/session/stateful/bm/TxMDBMSBeans/BeanA`

|`ejb/ee/tx/session/stateful/bm/TxMDBMSBeans/BeanB`
|`ejb/ee/tx/session/stateful/bm/TxMDBMSBeans/BeanC`

|`ejb/ee/tx/session/stateful/bm/TxMDBSS_Direct`
|`ejb/ee/tx/session/stateful/bm/TxMDBSSBeans/BeanA`

|`ejb/ee/tx/session/stateful/bm/TxMDBSSBeans/BeanB`
|`tests/ejb/ee/tx/session/stateful/bm/TxMDBSSBeans/BeanC`

|`ejb/ee/tx/session/stateful/bm/TxMDBSS_Indirect`
|`ejb/ee/tx/session/stateful/bm/TxMDBSSBeans/BeanA`

|`ejb/ee/tx/session/stateful/bm/TxMDBSSBeans/BeanB`
|`ejb/ee/tx/session/stateful/bm/TxMDBSSBeans/BeanC`
```

Comment (scottmarlow) [reply src/com/sun/ts/lib/deliverable/cts/deploy/DeployTestUtil.java]: I am thinking that the Common Applications Deployment will not work without Jakarta Deployment.  If that is the case, we should remove the Common Applications Deployment tests, docs, support classes.  In theory, we should introduce an alternative to Jakarta Deployment as discussed but we don't have that in place yet, so likely need to remove CAD.

Comment (scottmarlow) [reply src/com/sun/ts/lib/deliverable/cts/deploy/DeployTestUtil.java]: Updated plan, to try to get through the build with failures, will remove Common Applications Deployment in a separate pull request.

**Comment by scottmarlow** [comment]
Will merge this now, further feedback is still appreciated! As mentioned above we still expect to remove the Common Application Deployment in a separate pull request.

File: install/jakartaee/bin/ts.jte [hunk 2]
```diff
@@ -1581,56 +1566,30 @@ porting.ts.deploy.class.1=com.sun.ts.lib.implementation.sun.javaee.glassfish.Aut
...
-#                      process for the RI
-# @deployManageruname.2  username for deployment process for the RI
-# @deployManagerpasswd.2  password for deployment process for the RI
+## Following are no longer used, were settings for pruned Jakarta Deployment
 ########################################################################
-deployManagerJarFile.2=${javaee.home.ri}/lib/deployment/sun-as-jsr88-dm.jar
-deployManageruri.2=deployer:Sun:AppServer::${deployment_host.2}:${ri.admin.port}
-deployManageruname.2=${ri.admin.user}
-deployManagerpasswd.2=${ri.admin.passwd}
+#deployManagerJarFile.2
```
Comment (alwin-joseph) [inline install/jakartaee/bin/ts.jte:1583]: Property fetching causing exception:

As of now there are exceptions while sample test run as below :
" com.sun.ts.lib.deliverable.PropertyNotSetException: No value found for deployManagerJarFile.1.  Please check your jte file for an appropiate value " from ./src/com/sun/ts/lib/deliverable/AbstractPropertyManager.java
So looks like it does not allow null for these currently when fetched at src/com/sun/ts/lib/deliverable/cts/CTSPropertyManager.java

Comment (scottmarlow) [reply install/jakartaee/bin/ts.jte:1583]: Thanks @alwin-joseph, I will look at this.

Comment (scottmarlow) [reply install/jakartaee/bin/ts.jte:1583]: Push some possible changes to https://github.com/scottmarlow/jakartaee-tck/tree/moreDeploymentRemoval

Comment (scottmarlow) [reply install/jakartaee/bin/ts.jte:1583]: We are now getting ```execStatus=Failed. Illegal access to test\: java.lang.NullPointerException```

Comment (scottmarlow) [reply install/jakartaee/bin/ts.jte:1583]: @alwin-joseph from the ./root/vi/glassfish6/glassfish/domains/domain1/logs/server.log from [samples-results.tar.gz](https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck/job/master/678/artifact/samples-results.tar.gz):
```
[2020-06-08T17:09:01.779+0000] [glassfish 6.0] [INFO] [NCLS-DEPLOYMENT-02027] [javax.enterprise.system.tools.deployment.autodeploy] [tid: _ThreadID=54 _ThreadName=AutoDeployer] [timeMillis: 1591636141779] [levelValue: 800] [[
  Selecting file /root/vi/glassfish6/glassfish/domains/domain1/autodeploy/ejb_sam_Hello.ear for autodeployment]]
...
[2020-06-08T17:09:03.332+0000] [glassfish 6.0] [SEVERE] [NCLS-CORE-00029] [javax.enterprise.system.core] [tid: _ThreadID=54 _ThreadName=AutoDeployer] [timeMillis: 1591636143332] [levelValue: 1000] [[
  Cannot start container ejb, exception: A MultiException has 7 exceptions.  They are:
1. com.sun.enterprise.module.ResolveError: Failed to start OSGiModuleImpl:: Bundle = [org.glassfish.main.orb.iiop [218]], State = [NEW]
2. java.lang.IllegalStateException: Could not load descriptor SystemDescriptor(
        implementation=org.glassfish.enterprise.iiop.impl.GlassFishORBFactoryImpl
        contracts={org.glassfish.enterprise.iiop.impl.GlassFishORBFactoryImpl,org.glassfish.enterprise.iiop.api.GlassFishORBFactory}
        scope=jakarta.inject.Singleton
        qualifiers={}
        descriptorType=CLASS
        descriptorVisibility=NORMAL
        metadata=Bundle-SymbolicName={org.glassfish.main.orb.iiop},Bundle-Version={6.0.0.SNAPSHOT}
        rank=0
        loader=OsgiPopulatorPostProcessor.HK2Loader(OSGiModuleImpl:: Bundle = [org.glassfish.main.orb.iiop [218]], State = [NEW],655682992)
        proxiable=null
        proxyForSameScope=null
        analysisName=null
        id=110
        locatorId=0
        identityHashCode=444705614
        reified=false)
3. java.lang.IllegalStateException: Unable to perform operation: post construct on org.glassfish.enterprise.iiop.api.GlassFishORBHelper
4. java.lang.IllegalArgumentException: While attempting to resolve the dependencies of com.sun.ejb.containers.EjbContainerUtilImpl errors were found
5. java.lang.IllegalStateException: Unable to perform operation: resolve on com.sun.ejb.containers.EjbContainerUtilImpl
6. java.lang.IllegalArgumentException: While attempting to resolve the dependencies of org.glassfish.ejb.startup.EjbContainerStarter errors were found
7. java.lang.IllegalStateException: Unable to perform operation: resolve on org.glassfish.ejb.startup.EjbContainerStarter
```
NPE call stack is
```
[runcts] OUT => [javatest.batch] java.lang.NullPointerException
   [runcts] OUT => [javatest.batch]     at com.sun.ts.lib.harness.TSScript.run(TSScript.java:320)
   [runcts] OUT => [javatest.batch]     at com.sun.javatest.Script.run(Script.java:241)
   [runcts] OUT => [javatest.batch]     at com.sun.javatest.DefaultTestRunner.runTest(DefaultTestRunner.java:174)
   [runcts] OUT => [javatest.batch]     at com.sun.javatest.DefaultTestRunner.access$100(DefaultTestRunner.java:43)
   [runcts] OUT => [javatest.batch]     at com.sun.javatest.DefaultTestRunner$1.run(DefaultTestRunner.java:66)
```

Comment (scottmarlow) [reply install/jakartaee/bin/ts.jte:1583]: Created minor NPE fix via https://github.com/eclipse-ee4j/jakartaee-tck/pull/311.

Comment (alwin-joseph) [reply install/jakartaee/bin/ts.jte:1583]: > Pushed some possible changes to https://github.com/scottmarlow/jakartaee-tck/tree/moreDeploymentRemoval

This issue is fixed now in the latest build. Thanks.

Comment (alwin-joseph) [reply install/jakartaee/bin/ts.jte:1583]: > Created minor NPE fix via #311.

Scott - From what I see the root problem is that the deployment fails due to some reason. I am not sure if the fix is addressing that. The deployment failures are there for several packages even for standalone tck runs and need to be looked into. We can use the tck bundles built and try them in local to check the server logs. It is also possible something is wrong with the deployment rar/war or with glassfish itself.

Comment (alwin-joseph) [reply install/jakartaee/bin/ts.jte:1583]: From the server logs :

[2020-06-08T16:58:42.522+0000] [glassfish 6.0] [WARNING] [] [] [tid: _ThreadID=52 _ThreadName=AutoDeployer] [timeMillis: 1591635522522] [levelValue: 900] [[
  Error occurred
org.xml.sax.SAXParseException; lineNumber: 52; columnNumber: 106; Deployment descriptor file META-INF/ra.xml in archive [whitebox-tx.rar].  cvc-enumeration-valid: Value 'jakarta.resource.spi.security.PasswordCredential' is not facet-valid with respect to enumeration '[javax.resource.spi.security.PasswordCredential, org.ietf.jgss.GSSCredential, javax.resource.spi.security.GenericCredential]'. It must be a value from the enumeration.

I see glassfish6/glassfish/lib/schemas/connector_1_7.xsd & glassfish6/glassfish/lib/dtds/connector_1_5.dtd having javax.resource.spi.security.PasswordCredential. I suspect these are to be changed to jakarta.* in glassfish ?

Comment (scottmarlow) [reply install/jakartaee/bin/ts.jte:1583]: Good catch, yes, I agree that glassfish should update these.
I did find https://github.com/eclipse-ee4j/glassfish/issues/23080 which is for the ORB failure, that we also need fixed I think.

Links for updated XSDs:
https://jakarta.ee/xml/ns/jakartaee/connector_2_0.xsd
https://github.com/eclipse-ee4j/jakartaee-schemas/blob/master/jakartaee9/src/connector_2_0.xsds

It looks like we already updated jakartaee-tck/src/com/sun/ts/tests/common/connector/whitebox/permissiondd/ra.xml, which is good.

I'm not sure if any of the TCK lib/schemas are actually used by the Platform TCK but we should update any of those that are needed, to use their EE 9 equivalents (as per my discussion with legal@eclipse.org).  I think the same for lib/dtds.

**Pull request merged**